src/Microsoft.Health.Fhir.Core/Features/Search/SearchIndexer.cs

```diff
@@ -227,5 +227,30 @@ private IEnumerable<SearchIndexEntry> ProcessNonCompositeSearchParameter(SearchP
 
             return results;
         }
+
+        private static Type GetSearchValueTypeForSearchParamType(SearchParamType searchParamType)
```

Comment:
We could also use attribute and regsiter the type against derived class instead of manually map it here (but i think this will need to change a bit with the R4 work anyway so we can see how it evolve later).

Reply:
To be revisited once R4 lands.